pkgs/shells/powershell/default.nix

```diff
@@ -0,0 +1,87 @@
+{ stdenv, fetchgit, dotnet-sdk, autoPatchelfHook, fetchzip, libunwind, libuuid, icu, git, curl, cmake, cacert,
+  glibcLocales, makeWrapper, less, openssl }:
+
+let bootstrap = vsn: stdenv.mkDerivation rec {
+  name = "powershell-bootstrap-${version}";
+  version = vsn;
+
+  src = fetchzip {
+    url = "https://github.com/PowerShell/PowerShell/releases/download/v${version}/powershell-${version}-linux-x64.tar.gz";
+    sha256 = "000mmv5iblnmwydfdvg5izli3vpb6l14xy4qy3smcikpf0h87fhl";
+    stripRoot = false;
+  };
+
+  buildInputs = [ autoPatchelfHook stdenv.cc.cc ];
+
+  installPhase = ''
+  mkdir -p $out/bin
+  cp -r * $out/bin
+  touch a
+  '';
+
+  dontStrip = true;
+
+
+}; in
+stdenv.mkDerivation rec {
+  name = "powershell-${version}";
+  version = "6.0.2";
+  
+  src = fetchgit {
+    url = "https://github.com/PowerShell/PowerShell.git";
+    rev = "v${version}";
+    sha256 = "0hhh33cdfs29gy73p6bp56nqqv3cr3x9r3j7ib6141arp5yn0kxf";
+    leaveDotGit = true;
+    fetchSubmodules = true;
+    deepClone = true;
```

Comment:
Do you really need a deep clone here?

Reply:
Without `deepClone`, I get this error during the build:

```
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.Commands.Management/Microsoft.PowerShell.Commands.Management.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.Commands.Utility/Microsoft.PowerShell.Commands.Utility.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.ConsoleHost/Microsoft.PowerShell.ConsoleHost.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.Security/Microsoft.PowerShell.Security.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/System.Management.Automation/System.Management.Automation.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.CoreCLR.Eventing/Microsoft.PowerShell.CoreCLR.Eventing.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.SDK/Microsoft.PowerShell.SDK.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/Microsoft.PowerShell.PSReadLine/Microsoft.PowerShell.PSReadLine.csproj]
/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/PowerShell.Common.props(17,5): error MSB3073: The command "git describe --abbrev=60 --long" exited with code 128. [/tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/src/powershell-unix/powershell-unix.csproj]
Execution of { dotnet restore $_ $RestoreArguments } by build.psm1: line 536 failed with exit code 1
At /tmp/nix-build-powershell-6.0.2.drv-0/PowerShell/build.psm1:2054 char:17
+                 throw $errorMessage
+                 ~~~~~~~~~~~~~~~~~~~
+ CategoryInfo          : OperationStopped: (Execution of { ...ith exit code 1:String) [], RuntimeException
+ FullyQualifiedErrorId : Execution of { dotnet restore $_ $RestoreArguments } by build.psm1: line 536 failed with exit code 1
```

Reply:
@Mic92 I pushed an update to avoid this dependence on `deepClone` -- but I still depend on `.git` to pick the correct version with `git rev-parse HEAD` to make everything look perfect to powershell build system just in case. Would this work?